docs/frontend_architecture/unit_testing.rst

```diff
@@ -20,5 +19,77 @@ Tests are written in JavaScript, and placed in the 'assets/test' folder. An exam
+
+  describe('Heading', () => {
+    it('renders a heading', async () => {
+      render(Heading, {
+        props: {
+          text: 'Hello, world!',
+        },
+      });
+
+      expect(screen.getByRole('heading')).toHaveTextContent('Hello, world!');
+    });
+  });
+
+
+We use `Vue Testing Library (VTL) <https://testing-library.com/docs/vue-testing-library/intro/>`__ and its associated ecosystem to test Vue components, as it allows us to test UI components in a user-centric way.
+
+It is based on the philosophy that “The more your tests resemble the way your software is used, the more confidence they can give you." Rather than dealing with instances of rendered Vue components, it allows our tests to work with actual DOM nodes and simulate interactions the same way the user would. We earlier made use of `Vue Test Utils <https://v1.test-utils.vuejs.org/>`__ for the frontend testing, but have been transitioning to VTL for the same.
+
+To learn more about VTL, you can check it's `examples page <https://testing-library.com/docs/vue-testing-library/examples>`__. You can also check out our `testing templates <testing_layout.html>`__ and our style guide to start writing new test suites in Kolibri.
+
+Style Guide
+-----------
+
+Here are some of the conventions that we prefer to follow while testing so that uniformity is maintained across the codebase:
+
+Naming Conventions
+~~~~~~~~~~~~~~~~~~
+
+Each folder is expected to have a ``__tests__`` folder, which would contain all the test files for the files in the same (whether ``.js`` or ``.vue``). The test files are expected to follow the naming convention ``<Name of the file being tested>.spec.js``.
+
+Use of ``renderComponent`` function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To avoid repeating boilerplate code while testing Vue components, define a ``renderComponent`` function to set up all the necessary mocks, stubs, and default props values to render the component to test. It can also revieve an optional argument ``props``, which can be used to overwrite:
+
+-  The default props passed to the component being rendered
+-  Configuration passed to other mocks/stubs (like the values the getters for store mock should return, arguments to Vue Router etc) according to the test case
+
+Queries
+~~~~~~~
+
+VTL provides a number of `queries <https://testing-library.com/docs/vue-testing-library/cheatsheet#queries>`__ that can be used to query the DOM nodes. There are primarily three types of queries: ``get``, ``query`` and ``find``. All of these queries have different variants, which are used to query the DOM nodes based on different criteria. Some examples of the same include: ``getByText``, ``queryByRole``, ``findByText`` etc. These queries also have a recommened priority based on what the user would most likely interact with. You can read more about the same `here <https://testing-library.com/docs/queries/about#priority>`__.
+
+Making use of ``screen``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I would change this to something like:

```rst
Making use of VTL ``screen`` object
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For querying DOM nodes, the preferred way is to use the ``screen`` object provided by ``@testing-library/vue``. So instead of destructuring the queries functions from the object returned by the VTL ``render`` function, you can use the VTL ``screen`` object for faster access to all its queries:

.. code-block:: javascript
// ❌
const {getByRole} = render(Example)
const errorMessage = getByRole('alert')

// ✅
render(<Example />)
const errorMessage = screen.getByRole('alert')
```

I think we can write some examples here to make it clearer what we want to communicate.

Reply:
Got it! This is a pretty neat way to communicate the same. Thanks!